docker/docker.go

```diff
@@ -24,11 +24,118 @@ const (
 	defaultCertFile     = "cert.pem"
 )
 
+func checkDaemon(args []string) bool {
+	for j := 1; j < len(args); j++ {
+		if args[j] == "-d" || args[j] == "--daemon" {
+			return true
+		} else if args[j][:1] == "-" {
+			if index_of_equal := strings.Index(args[j], "="); index_of_equal != -1 {
+				continue //Flag with a value
+			} else {
+				isBool, exists := flag.CommandLine.FlagsMap[args[j]]
+				if exists && !isBool {
+					//Global string flag
+					j = j + 1
+					continue
+				} else if exists && isBool {
+					//Global bool flag
+					continue
+				}
+
+				isBool, exists = cmd.FlagsMap[args[j]]
+				if exists && !isBool {
+					//Daemon string flag
+					j = j + 1
+					continue
+				} else if exists && isBool {
+					//Daemon bool flag
+					continue
+				}
+			}
+		} else {
+			for _, command := range Commands {
+				if args[j] == command[0] {
+					return false
+				}
+			}
+		}
+	}
+	return false
+}
+
+func shuffle(args []string) []string {
+	var (
+		global_args = []string{args[0]}
+		rest_args   = []string{"daemon"}
+	)
+
+	logrus.Printf("-d and --daemon are deprecated. Please use daemon command")
+
+	//Shuffling of arguments required.Create global and rest args list.
+
+	for j := 1; j < len(args); j++ {
+		if args[j] != "-d" && args[j] != "--daemon" {
+			if args[j][:1] == "-" {
+				if index_of_equal := strings.Index(args[j], "="); index_of_equal != -1 {
+					_, exists := flag.CommandLine.FlagsMap[args[j][:index_of_equal]]
+					if exists {
+						//Global flag with a value
+						global_args = append(global_args, args[j])
+						continue
+					}
+					_, exists = cmd.FlagsMap[args[j][:index_of_equal]]
+					if exists {
+						//Daemon flag with a value
+						rest_args = append(rest_args, args[j])
+						continue
+					}
```

Comment:
We would need both maps, as those maps are populating which flags are boolean and which are not, thereby identifying which can take an optional argument. One way is to hard code all the flags in one single map. In order to avoid that we are populating maps as part of flagset. Since we have segregated the flagsets {daemon, global} two maps are dynamically generated for them.